src/core/control.pm

```diff
@@ -215,6 +215,10 @@ proto sub EVAL(Cool $code, Str() :$lang = 'perl6', PseudoStash :$context, *%n) {
     nqp::forceouterctx(nqp::getattr($compiled, ForeignCode, '$!do'), $eval_ctx);
     $compiled();
 }
+multi sub EVAL(Cool $code, Str() :$lang! where 'nqp', *%_) {
```

Comment:
the where should probably follow the sub below with `$lang // ''`? or maybe it should be changed to use `:$lang!`?